python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_defs.py

```diff
@@ -0,0 +1,60 @@
+import re
+from typing import Any, Dict, List, Optional, Union
+
+from dagster import (
+    AssetExecutionContext,
+    AssetsDefinition,
+    AssetSpec,
+    MaterializeResult,
+    multi_asset,
+)
+
+from dagster_embedded_elt.sling.resources import SlingMode, SlingResource
+
+
+def build_sling_asset(
+    asset_spec: AssetSpec,
+    source_stream: str,
+    target_object: str,
+    mode: SlingMode = SlingMode.FULL_REFRESH,
+    primary_key: Optional[Union[str, List[str]]] = None,
+    update_key: Optional[Union[str, List[str]]] = None,
+    source_options: Optional[Dict[str, Any]] = None,
```

Comment:
eventually, it would be nice to get these type hinted w/ Pydantic - I think it's fine for now to have an unstructured dict